p2p/router.go

```diff
@@ -238,11 +238,12 @@ func NewRouter(
 // close the channel when done, before stopping the Router. messageType is the
 // type of message passed through the channel (used for unmarshaling), which can
 // implement Wrapper to automatically (un)wrap multiple message types in a
-// wrapper message.
-func (r *Router) OpenChannel(id ChannelID, messageType proto.Message) (*Channel, error) {
-	queue := newFIFOQueue()
-	outCh := make(chan Envelope)
-	errCh := make(chan PeerError)
+// wrapper message. The caller may provide a size to make the channel buffered,
+// which internally makes the inbound, outbound, and error channel buffered.
+func (r *Router) OpenChannel(id ChannelID, messageType proto.Message, size int) (*Channel, error) {
+	queue := newFIFOQueue(size)
+	outCh := make(chan Envelope, size)
+	errCh := make(chan PeerError, size)
```

Comment:
/cc @erikgrinaker -- in order to get this to play nicely with the p2p test network stack and the consensus reactor, we need these channels to be buffered. In the router, of course, we maintain them to be unbuffered (i.e. 0).

Not sure if you had thoughts on this as far as APIs and what should be buffered, how much, and where.

Reply:
Sure. You could consider buffering the `MemoryConnection` queues instead, since all networked transports (including `MConnTransport`) does a bunch of internal buffering already, as does the OS networking stack, so this gives more consistent behavior across transports.

That said, any non-trivial `queue` implementations will need internal buffers too (because they need a set of messages to apply QoS algorithms to).

Reply:
> That said, any non-trivial queue implementations will need internal buffers too (because they need a set of messages to apply QoS algorithms to).

This encourages me to leave this as-is for now. Although I do agree we should allow the `MemoryConnection` to be configurable.